spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/function/FunctionDetectorCondition.java

```diff
@@ -101,11 +110,31 @@ public ConditionOutcome getMatchOutcome(ConditionContext context, AnnotatedTypeM
 						prunedList.add(key);
 					}
 				}
+				else {
+					//check if its a @Component bean.
+					Optional<Method> componentBeanMethod = Arrays.stream(methods).filter(
+							m -> m.getName().equals("apply") && isKafkaStreamsTypeFound(m) ||
+									m.getName().equals("accept") && isKafkaStreamsTypeFound(m)).findFirst();
```

Comment:
Should it be simpler `m -> (m.getName().equals("apply") || m.getName().equals("accept")) && isKafkaStreamsTypeFound(m)`